### ISSUE TITLE
Docker and pygame Display issue with Nvidia Drivers

### ISSUE THREAD

**Issue by jaskarannagi19** [pygame/pygame#2552]
I have been working with docker, nvidia gpus and Display before in docker image. Inside which I am successfully able to initialize pygame.init()

with output (7,0)

After formatting my ubuntu machine. I installed docker and nvidia container toolkit to enable gpus inside the docker container.

however after I use xhost + and run following docker command and bash to running container.

docker run -it -e DISPLAY=$DISPLAY --gpus all <IMAGE ID>

Inside the container. I can see nvidia-smi working and nvcc --version working. And installed apt-get install -qqy x11-apps. However when I install pygame using pip install pygame

And initialize pygame with pygame.init() it throws me an error:

pygame.error: No available video device

I have already gone though: pygame.error: No available video device

and many other link.

Would like to know if I am missing any libraries. On host machine or inside docker container and for reference I am using following container

docker run -it -e DISPLAY=$DISPLAY --gpus all nvcr.io/nvidia/tensorflow:20.06-tf2-py3

Other commands which I already have tried thanks to Stackoverflow community are following docker run -it -e DISPLAY=$DISPLAY -v SOME UNIX SOCKET MOUNT --gpus all nvcr.io/nvidia/tensorflow:20.06-tf2-py3

How do I connect my display to pygame so pygame.init() or pygame.display.init() does not break inside docker container.

Note: Python version 3

UPDATE 1: I tried without gpus all option

docker run -it -e DISPLAY=$DISPLAY nvcr.io/nvidia/tensorflow:20.06-tf2-py3

However it still does not work

**Comment by jaskarannagi19** [comment]
Hi @robertpfeiffer 
Can you please run 

docker run -it -e DISPLAY=$DISPLAY --gpus all nvcr.io/nvidia/tensorflow:20.06-tf2-py3

This command and try to init pygame inside the container?

Make sure docker and nvidia-docker2 is installed otherwise. Let me know what inputs are needed 

Thanks :)

**Comment by robertpfeiffer** [comment]
> Hi @robertpfeiffer
> Can you please run
> 
> docker run -it -e DISPLAY=$DISPLAY --gpus all nvcr.io/nvidia/tensorflow:20.06-tf2-py3
> 
> This command and try to init pygame inside the container?
> 
> Make sure docker and nvidia-docker2 is installed otherwise. Let me know what inputs are needed
> 
> Thanks :)

No. I am not going to debug your drivers/docker/X11 setup. I tagged this as needs-op-input, which means the maintainers won't be able to help you until you actually post what your problem is. Then you @mentioned me. So at the the very minimum you need to post....

...I am not going to tell you _again_ what to post. It was there in the new issue template, which you chose to delete and ignore. Post that. Additionally, you should know that some of these relevant information we need is from inside the container, and some information is from outside the container, so you need to run some diagnostics inside your container and some outside, and I am not going to tell you which is which. If you don't know enough about docker, maybe you need to read up on that first. Additionally, we probably need some information about which kind of setup (X11 or wayland+xwayland) you have on the outside.

This is not a docker help forum. This is also not a CUDA help forum, or a nvidia binary driver help forum. So please post the information we need to diagnose if this is actually an issue with pygame.
